sklearn/dummy.py

```diff
@@ -156,7 +157,10 @@ def fit(self, X, y, sample_weight=None):
 
         self.n_outputs_ = y.shape[1]
 
-        check_consistent_length(X, y, sample_weight)
+        check_consistent_length(X, y)
+
+        if sample_weight is not None:
+            sample_weight = _check_sample_weight(sample_weight, X)
```

Comment:
Actually it looks like several PRs were done for this estimator as this was added on the line below in https://github.com/scikit-learn/scikit-learn/pull/15505. Please remove the above 3 lines.

Otherwise LGTM.

Reply:
Is it worth merging at this point? Looks like the only changes left would be one linting issue and removing sample weight from check_consistent_length. Does it make more sense to close the PR instead?